x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/ExceptionsHelper.java

```diff
@@ -99,4 +99,8 @@ public static String shardFailuresToErrorMsg(String jobId, ShardSearchFailure[]
     public static <T> T requireNonNull(T obj, ParseField paramName) {
         return requireNonNull(obj, paramName.getPreferredName());
     }
+
+    public static Throwable unwrapCause(Throwable t) {
```

Comment:
What do you think of making this a `boolean unwrappedInstanceOf(Throwable wrappedThrowable, Class klass)` or something?

Reply:
Good idea! Will do.

Reply:
Having tried this I actually think it's harder to read.

It'd be for say `ResourceNotFoundException`:

```
if (ExceptionsHelper.unwrappedInstanceOf(e, ResourceNotFoundException.class)) {
  ...
}
```

instead of

```
if (ExceptionsHelper.unwrapCause(e) instanceof ResourceNotFoundException) {
  ...
}
```

My intuition says we'd read the latter more easily so I'll stick to it.